ModelBase/src/nodes/EmptyNode.h

```diff
@@ -0,0 +1,59 @@
+/***********************************************************************************************************************
+**
+** Copyright (c) 2015 ETH Zurich
+** All rights reserved.
+**
+** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+** following conditions are met:
+**
+**    * Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+**      disclaimer.
+**    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+**      following disclaimer in the documentation and/or other materials provided with the distribution.
+**    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+**      derived from this software without specific prior written permission.
+**
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+***********************************************************************************************************************/
+
+#pragma once
+
+#include "TypedList.h"
+#include "nodeMacros.h"
+
+DECLARE_TYPED_LIST(MODELBASE_API, Model, EmptyNode)
+
+namespace Model {
+
+/**
+ * EmptyNode can be used to create an empty visualization in a view, where
+ * it is required for the visualization to contain a node. This node should not
+ * actually be included in an AST.
+ */
+class MODELBASE_API EmptyNode: public Super<Node>
```

Comment:
This node, as you have defined it, is not empty at all. Rather it contains a QPoint. You must define the node in a different way. Instead of inheriting from Node directly, inherit from CompositeNode and use its mechanisms to provide the necessary attributes. For example see https://github.com/dimitar-asenov/Envision/blob/development/VisualizationBase/src/nodes/TestBoxNode.h and the corresponding .cpp file for an inspiration on how to implement your node.
- One of the properties of your node should be a reference node, which should be optional.